context/TokenData/TokenDataProvider.tsx

```diff
@@ -0,0 +1,201 @@
+  { "text_signature": "event AddLiquidity(address indexed provider, uint256 indexed eth_amount, uint256 indexed token_amount)" },
+  { "text_signature": "event RemoveLiquidity(address indexed provider, uint256 indexed eth_amount, uint256 indexed token_amount)" },
+  //IUniswapV2Pair
+  { "text_signature": "event Mint(address indexed sender, uint amount0, uint amount1)" },
+  { "text_signature": "event Burn(address indexed sender, uint amount0, uint amount1, address indexed to)" },
+  { "text_signature":
+    "event Swap(address indexed sender, uint amount0, uint amount1, uint amount0Out, uint amount1Out, address indexed to)" },
+  { "text_signature": "event Sync(uint112 reserve0, uint112 reserve1)" }
+]
+
+
+// const USDC = {
+//   "chainId": 1,
+//   "address": "0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48",
+//   "name": "USD Coin",
+//   "symbol": "USDC",
+//   "decimals": 6,
+//   "logoURI": "https://assets.coingecko.com/coins/images/6319/thumb/USD_Coin_icon.png?1547042389"
+// };
+
+const uniswapV2GQL = "https://api.thegraph.com/subgraphs/name/uniswap/uniswap-v2"
+const ethQL = `{
+  bundles (first:1) {
+    ethPrice
+  }
+}`
+
+
+type TokenDataContextProps = {
+  children: React.ReactNode | React.ReactNode[]
+}
+
+// TODO Update
+export type Token = {
+  address: string
+  coin: string
+  logo: string
+  decimals: number
+}
+
+export type Log = {
+
+}
+
+interface ITokenDataContextProps {
+  tokens: Token[]
+  getReceipts: (transaction: Record<string, unknown>) => Promise<Log[]>
+}
+
+
+const TokenDataContext = React.createContext<ITokenDataContextProps | undefined>(undefined)
+
+const TokenDataProvider = ({ children }: TokenDataContextProps) => {
+  const [tokens, setTokens] = useState<Token[]>([])
+
+  const loadTokens = useCallback(async () => {
+    DEXES.map(d => {
+      // eslint-disable-next-line @typescript-eslint/no-var-requires
+      const { tokens } = require(`./tokensList/json${d}.json`)
+      console.log(tokens)
+      setTokens(tokens)
+    })
+  }, [])
+
+  const addEvents = useCallback(async () => {
+    eventsJson.map(async e => {
+      const { text_signature } = e
+      try {
+        const i = new Interface([text_signature])
+        await addABI(i.fragments)
+      } catch (e) {
+        console.log(e)
+      }
+    })
+  }, [])
+
+  const getEthPrice = async () =>  {
+    const res = await fetch(uniswapV2GQL, {
+      method: "POST",
+      headers: {
+        "Accept": "api_version=2",
+        "Content-Type": "application/graphql"
+      },
+      body: JSON.stringify({ query : ethQL })
+    })
+
+    const { data: { bundles } } = await res.json()
+
+    if (bundles.length > 0) {
+      return parseFloat(bundles[0].ethPrice).toFixed(6)
+    }
+
+    return "1"
+  }
+
+  const getAllLogs = useCallback(async (_logs)  => {
+    const ethPrice = await getEthPrice()
+    return decodeLogs(_logs).map(log => {
+      const { coin, logo, decimals } = tokens.find(token => token.address === log.address)
```

Comment:
this explodes when no token is found

![image](https://user-images.githubusercontent.com/1703831/123530590-97038a80-d6d2-11eb-8c7a-fb495afbb0d7.png)

Reply:
I didn't make any changes to the business logic here, what situation results in the no token error?